platform/admin/admin-configuration-configinstaller/src/main/java/org/codice/ddf/admin/configuration/ConfigurationInstaller.java

```diff
@@ -0,0 +1,201 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.admin.configuration;
+
+import static org.osgi.service.cm.ConfigurationEvent.CM_DELETED;
+import static org.osgi.service.cm.ConfigurationEvent.CM_UPDATED;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.codice.ddf.admin.configurator.Configurator;
+import org.codice.ddf.admin.configurator.ConfiguratorFactory;
+import org.codice.ddf.admin.configurator.Operation;
+import org.codice.ddf.admin.configurator.OperationReport;
+import org.codice.ddf.admin.configurator.Result;
+import org.codice.ddf.internal.admin.configurator.actions.ConfigActions;
+import org.osgi.framework.InvalidSyntaxException;
+import org.osgi.service.cm.Configuration;
+import org.osgi.service.cm.ConfigurationAdmin;
+import org.osgi.service.cm.ConfigurationEvent;
+import org.osgi.service.cm.SynchronousConfigurationListener;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Addresses shortcomings of the Felix {@code ConfigInstaller} with our own implementation that properly
+ * syncs the etc directory using the {@code felix.fileinstall.filename} property. Handling a config delete
+ * is fairly straight forward. The following outcomes are possible when an update occurs:
+ * <p>
+ * <table>
+ * <tr>
+ * <td>felixFileOriginal</td><td>felixFileNew</td><td>comments</td>
+ * </tr>
+ * <tr>
+ * <td>null</td><td>null</td><td>Felix file prop wasn't tracked and is still not tracked</td>
+ * </tr>
+ * <tr>
+ * <td>null</td><td>X</td><td>New config with a new Felix file prop to track</td>
+ * </tr>
+ * <tr>
+ * <td>Y</td><td>null</td><td>Felix file prop no longer being tracked, delete config file and remove from map</td>
+ * </tr>
+ * <tr>
+ * <td>Y</td><td>X</td><td>Felix file prop updated, delete config file and update map</td>
+ * </tr>
+ * <tr>
+ * <td>Y</td><td>Y</td><td>Felix file prop did not change, do nothing</td>
+ * </tr>
+ * </table>
+ *
+ * @see #handleUpdate(FelixConfig)
+ * @see #handleDelete(String)
+ */
+public class ConfigurationInstaller implements SynchronousConfigurationListener {
```

Comment:
✏️ My assumption is that the OSGi implementation is going to handle `SynchronousConfigurationListener`s in a _synchronous_ fashion, but that doesn't necessarily imply it won't concurrently invoke them. Your `pidFileMap` is definitely not thread-safe.

Unless you can definitively determine that implementations of this interface are never invoked concurrently by multiple threads, I suggest using a thread-safe map or guarding accesses to it.